src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1261,16 +1262,25 @@ def validate_ssh_key(namespace, cmd=None):
 
         processed_ssh_key_values = []
         for ssh_key_value in namespace.ssh_key_value:
-            processed_ssh_key_values.append(_validate_ssh_key_helper(ssh_key_value, namespace.generate_ssh_keys))
+            processed_ssh_key_values.append(_validate_ssh_key_helper(ssh_key_value,
+                                                                     namespace.generate_ssh_keys,
+                                                                     ssh_key_type))
         namespace.ssh_key_value = processed_ssh_key_values
     # if no ssh keys processed, try to generate new key / use existing at root.
     else:
-        namespace.ssh_key_value = [_validate_ssh_key_helper("", namespace.generate_ssh_keys)]
+        namespace.ssh_key_value = [_validate_ssh_key_helper("",
+                                                            namespace.generate_ssh_keys,
+                                                            ssh_key_type)]
 
 
-def _validate_ssh_key_helper(ssh_key_value, should_generate_ssh_keys):
-    string_or_file = (ssh_key_value or
-                      os.path.join(os.path.expanduser('~'), '.ssh', 'id_rsa.pub'))
+def _validate_ssh_key_helper(ssh_key_value, should_generate_ssh_keys, ssh_key_type=None):
+    if ssh_key_type is None or ssh_key_type == 'RSA':
+        string_or_file = (ssh_key_value or
+                          os.path.join(os.path.expanduser('~'), '.ssh', 'id_rsa.pub'))
+    else:
+        string_or_file = (ssh_key_value or
+                          os.path.join(os.path.expanduser('~'), '.ssh', 'id_ed25519.pub'))
```

Comment:
Could we simplify this logic?
```suggestion
   file_path = 'id_rsa.pub' if ssh_key_type == 'RSA' else 'id_ed25519.pub'
   string_or_file = (ssh_key_value or
                          os.path.join(os.path.expanduser('~'), '.ssh', file_path))
```